### PR TITLE
pass an internal pod object to SCC admission control so it works

### PR DESCRIPTION
**Pull request by bparees**
https://bugzilla.redhat.com/show_bug.cgi?id=1464356

bug 1464356

### CODE REVIEW

**Comment by bparees** [comment]
[test]

**Comment by bparees** [comment]
[testextended][extended:core(builds)]

**Comment by bparees** [comment]
@smarterclayton @deads2k ptal, i had to undo a bit of the v1 changes @smarterclayton made because the SCC admission controller only likes internal objects.

File: pkg/build/controller/strategy/sti.go [hunk 1]
```diff
@@ -112,27 +112,27 @@ func (bs *SourceBuildStrategy) CreateBuildPod(build *buildapi.Build) (*v1.Pod, e
 func (bs *SourceBuildStrategy) canRunAsRoot(build *buildapi.Build) bool {
 	var rootUser int64
 	rootUser = 0
-	pod := &v1.Pod{
```
Comment (smarterclayton) [inline pkg/build/controller/strategy/sti.go]: Scary

**Comment by smarterclayton** [comment]
Did this fail closed or fail open?

**Comment by bparees** [comment]
Open :(

Hence my adding a new conformance test.

But i did wonder if it implies the admission controller is fundamentally
flawed. I think the answer is no, though.

Ben Parees | OpenShift

**Comment by smarterclayton** [comment]
We should probably fail closed in SCC on a known set of resources if
the object type is not recognized

**Comment by bparees** [comment]
> We should probably fail closed in SCC on a known set of resources if
the object type is not recognized

w/ the known set being the internal versions of all the resources in k8s+openshift?  doesn't the admission controller get called on every create of every resource, or is the chain smarter than that?  (that's why i decided the admission controller is probably right as is, because the alternative seems awkward and we really only ran into this situation because we're sort of abusing the admission controller)

**Comment by pweil-** [comment]
We already check for resource and subresource.  Failing closed on the conversion seems like the right thing to do here.  https://github.com/openshift/origin/pull/14901

**Comment by deads2k** [comment]
> We already check for resource and subresource. Failing closed on the conversion seems like the right thing to do here. #14901

Yeah, we should fail closed.  We already try to check if its a pod coming to us.

@bparees is there an issue open to switch the check APIs?

[merge]

**Comment by bparees** [comment]
> @bparees is there an issue open to switch the check APIs?

you mean switch the way we are checking if the user can run as root?  no, can you open one and put some details/pointers in about what we should be using instead (and then assign it to me).

**Comment by deads2k** [comment]
> you mean switch the way we are checking if the user can run as root? no, can you open one and put some details/pointers in about what we should be using instead (and then assign it to me).

Opened https://github.com/openshift/origin/issues/14909 with a link to the API type.

**Pull request merged**